docker/mongodb-kubernetes-tests/tests/authentication/replica_set_scram_sha_1_switch_project.py

```diff
@@ -0,0 +1,166 @@
+
+    def test_replica_set_connectivity(self):
+        """
+        Verify connectivity to the original replica set.
+        """
+        ReplicaSetTester(MDB_RESOURCE_NAME, 3).assert_connectivity()
+
+    def test_ops_manager_state_correctly_updated_in_initial_replica_set(self, replica_set: MongoDB):
+        """
+        Ensure Ops Manager state is correctly updated in the original replica set.
+        """
+        tester = replica_set.get_automation_config_tester()
+        tester.assert_authentication_mechanism_enabled("MONGODB-CR")
+        tester.assert_authoritative_set(True)
+        tester.assert_authentication_enabled(2)
+        tester.assert_expected_users(0)
+
+    def test_create_secret(self):
+        print(f"creating password for MongoDBUser {self.USER_NAME} in secret/{self.PASSWORD_SECRET_NAME} ")
+
+        create_or_update_secret(
+            KubernetesTester.get_namespace(),
+            self.PASSWORD_SECRET_NAME,
+            {
+                "password": self.USER_PASSWORD,
+            },
+        )
+
+    def test_create_user(self, namespace: str):
+        mdb = MongoDBUser.from_yaml(
+            load_fixture("scram-sha-user.yaml"),
+            namespace=namespace,
+        )
+        mdb["spec"]["mongodbResourceRef"]["name"] = MDB_RESOURCE_NAME
+
+        mdb.update()
+        mdb.assert_reaches_phase(Phase.Updated, timeout=150)
+
+    def test_ops_manager_state_with_users_correctly_updated(self, replica_set: MongoDB):
+        expected_roles = {
+            ("admin", "clusterAdmin"),
+            ("admin", "userAdminAnyDatabase"),
+            ("admin", "readWrite"),
+            ("admin", "userAdminAnyDatabase"),
+        }
+
+        tester = replica_set.get_automation_config_tester()
+        tester.assert_has_user(self.USER_NAME)
+        tester.assert_user_has_roles(self.USER_NAME, expected_roles)
+        tester.assert_expected_users(1)
+
+    def test_switch_replica_set_project(
+        self, custom_mdb_version: str, replica_set: MongoDB, namespace: str, project_name_prefix: str
+    ):
+        """
+        Modify the replica set to switch its Ops Manager reference to a new project and verify lifecycle.
+        """
+        original_configmap = read_configmap(namespace=namespace, name="my-project")
+        new_project_name = f"{project_name_prefix}-second"
+        new_project_configmap = create_or_update_configmap(
+            namespace=namespace,
+            name=new_project_name,
+            data={
+                CONFIG_MAP_KEYS["BASE_URL"]: original_configmap[CONFIG_MAP_KEYS["BASE_URL"]],
+                CONFIG_MAP_KEYS["PROJECT_NAME"]: new_project_name,
+                CONFIG_MAP_KEYS["ORG_ID"]: original_configmap[CONFIG_MAP_KEYS["ORG_ID"]],
+            },
+        )
+
+        replica_set.load()
+        replica_set["spec"]["opsManager"]["configMapRef"]["name"] = new_project_configmap
+        replica_set.set_version(custom_mdb_version)
```

Comment:
the version should be only set in the resource's fixture unless the changing version is part of the test.